server/protocols/wkd/src/main/java/org/apache/james/wkd/mailet/WebKeyDirectoryMailet.java

```diff
@@ -0,0 +1,185 @@
+package org.apache.james.wkd.mailet;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Iterator;
+
+import javax.inject.Inject;
+import javax.mail.BodyPart;
+import javax.mail.MessagingException;
+import javax.mail.Multipart;
+import javax.mail.internet.MimeMessage;
+
+import org.apache.james.domainlist.api.DomainList;
+import org.apache.james.wkd.crypto.WebKeyDirectorySubmissionAddressKeyPairManager;
+import org.apache.james.wkd.store.PublicKeyEntry;
+import org.apache.james.wkd.store.WebKeyDirectoryStore;
+import org.apache.mailet.Mail;
+import org.apache.mailet.base.GenericMailet;
+import org.bouncycastle.openpgp.PGPCompressedData;
+import org.bouncycastle.openpgp.PGPEncryptedDataList;
+import org.bouncycastle.openpgp.PGPException;
+import org.bouncycastle.openpgp.PGPLiteralData;
+import org.bouncycastle.openpgp.PGPObjectFactory;
+import org.bouncycastle.openpgp.PGPPrivateKey;
+import org.bouncycastle.openpgp.PGPPublicKey;
+import org.bouncycastle.openpgp.PGPPublicKeyEncryptedData;
+import org.bouncycastle.openpgp.PGPPublicKeyRing;
+import org.bouncycastle.openpgp.PGPUtil;
+import org.bouncycastle.openpgp.jcajce.JcaPGPObjectFactory;
+import org.bouncycastle.openpgp.jcajce.JcaPGPPublicKeyRingCollection;
+import org.bouncycastle.openpgp.operator.PublicKeyDataDecryptorFactory;
+import org.bouncycastle.openpgp.operator.bc.BcKeyFingerprintCalculator;
+import org.bouncycastle.openpgp.operator.bc.BcPublicKeyDataDecryptorFactory;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class WebKeyDirectoryMailet extends GenericMailet {
```

Comment:
We generaly write a piece of java doc explaining what the mailet does and also detailing how to configure it. A maven plugin then generate built in plugin documentation from the code.